tests/unit/test_charm_balancer.py

```diff
@@ -67,16 +85,17 @@ def test_stop_workload_if_not_leader(patched_restart, charm_configuration):
         config=charm_configuration,
         actions=ACTIONS,
     )
-    state_in = State(leader=False, relations=[])
+    state_in = base_state.replace(leader=False)
```

Comment:
**praise**:  TIL about the `replace` method. This is an excellent way to keep tests similar between k8s and vm